oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/NodeImpl.java

```diff
@@ -159,6 +161,16 @@
 
     public NodeImpl(T dlg, SessionContext sessionContext) {
         super(dlg, sessionContext);
+        logWarnStringSizeThreshold = getLogWarnStringSizeThreshold();
+    }
+
+    private int getLogWarnStringSizeThreshold() {
+    int threshold = OakJcrConstants.DEFAULT_WARN_LOG_STRING_SIZE_THRESHOLD_VALUE;
+        if (System.getProperty(OakJcrConstants.WARN_LOG_STRING_SIZE_THRESHOLD_KEY) != null
+                && !System.getProperty(OakJcrConstants.WARN_LOG_STRING_SIZE_THRESHOLD_KEY).isEmpty()) {
+            threshold = Integer.parseInt(System.getProperty(OakJcrConstants.WARN_LOG_STRING_SIZE_THRESHOLD_KEY));
```

Comment:
`Integer#parseInt` can throw `NumberFormatException` which should be caught an handled. But, better yet, it would cleaner to simply use `Integer#getInt(String, int)`.

Reply:
Done